File: crates/anvil-kernel/tests/fixtures/langtail/greeter.dart
// LANGTAIL T1 fixture — representative Dart source (Flutter-flavoured).
import 'dart:async';
import 'package:meta/meta.dart';

/// A small greeter service.
class Greeter {
  final String prefix;

  Greeter(this.prefix);

  Future<String> greet(String name) async {
    return '$prefix, $name';
  }

  void _logInternal(String message) {
    print(message);
  }
}

mixin Loggable {
  void log(String line) => print(line);
}

enum Mood { happy, neutral, grumpy }

String topLevelGreeting(String who) => 'hello $who';

File: crates/anvil-kernel/tests/fixtures/langtail/greeter.zig
const std = @import("std");

pub const Mood = enum { happy, sad };

pub const Greeter = struct {
    name: []const u8,

    pub fn greet(self: Greeter) []const u8 {
        return self.name;
    }

    fn secret(self: Greeter) void {
        _ = self;
    }
};

pub fn topLevelGreeting(g: Greeter) []const u8 {
    return g.greet();
}

fn internalHelper() void {}
